src/tests/test_sphincsplus.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+* (C) 2023 Jack Lloyd
+*     2023 Fabian Albert, René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "test_rng.h"
+#include "tests.h"
+
+#if(defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) || defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)) && defined(BOTAN_HAS_AES)
+
+   #include <botan/hash.h>
+   #include <botan/hex.h>
+
+   #include <botan/assert.h>
+   #include <botan/hash.h>
+   #include <botan/pk_algs.h>
+   #include <botan/pubkey.h>
+   #include <botan/secmem.h>
+   #include <botan/sp_parameters.h>
+   #include <botan/sphincsplus.h>
+   #include <botan/internal/loadstor.h>
+   #include <botan/internal/sp_hash.h>
+
+   #include "test_pubkey.h"
+
+namespace Botan_Tests {
+
+/**
+ * Test all implemented SPHINCS+ instances using the data of the KAT files
+ * generated by the reference implementation (Round 3.1
+ *  (https://github.com/sphincs/sphincsplus/))
+ */
+class SPHINCS_Plus_Test final : public Text_Based_Test {
+   public:
+      SPHINCS_Plus_Test() : Text_Based_Test("pubkey/sphincsplus.vec", "SphincsParameterSet,seed,pk,sk,msg,HashSig") {}
+
+      bool skip_this_test(const std::string&, const VarMap& vars) override {
+         [[maybe_unused]] auto params = Botan::Sphincs_Parameters::create(vars.get_req_str("SphincsParameterSet"));
+
+   #if not defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)
+         if(params.hash_type() == Botan::Sphincs_Hash_Type::Shake256) {
+            return true;
+         }
+   #endif
+
+   #if not defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2)
+         if(params.hash_type() == Botan::Sphincs_Hash_Type::Sha256) {
+            return true;
+         }
+   #endif
+
+         return false;
+      }
+
+      Test::Result run_one_test(const std::string&, const VarMap& vars) final {
+         Test::Result result("SPHINCS+ ");
+
+         auto params = Botan::Sphincs_Parameters::create(vars.get_req_str("SphincsParameterSet"));
+
+         const std::vector<uint8_t> seed_ref = vars.get_req_bin("seed");
+         const std::vector<uint8_t> msg_ref = vars.get_req_bin("msg");
+         const std::vector<uint8_t> pk_ref = vars.get_req_bin("pk");
+         const std::vector<uint8_t> sk_ref = vars.get_req_bin("sk");
+         const std::vector<uint8_t> sig_hash = vars.get_req_bin("HashSig");
+
+         // Depending on the SPHINCS+ configuration the resulting signature is
+         // hashed either with SHA-3 or SHA-256 to reduce the inner dependencies
+         // on other hash function modules.
+         auto hash = [&]() -> std::unique_ptr<Botan::HashFunction> {
+            if(params.hash_type() == Botan::Sphincs_Hash_Type::Shake256) {
+               return Botan::HashFunction::create("SHA-3(256)");
+            } else {
+               return Botan::HashFunction::create("SHA-256");
+            }
+         }();
```

Comment:
It might be clearer if the lambda just returned the string, and then we call `Botan::HashFunction::create` on the result of the lambda.

Reply:
Done. Fixed in 3dec39d